lib/cosmos/system/target.rb

```diff
@@ -131,16 +130,18 @@ def process_file(filename)
         when 'REQUIRE'
           usage = "#{keyword} <FILENAME>"
           parser.verify_num_parameters(1, 1, usage)
+          filename = File.join(@dir, 'lib', parameters[0])
           begin
-            # Require absolute path to file in target lib folder.  Prevents name
+            # Require absolute path to file in target lib folder. Prevents name
             # conflicts at the require step
             Cosmos.disable_warnings do
-              Cosmos.require_file(File.join(@dir, 'lib', parameters[0]), false)
+              Cosmos.require_file(filename, false)
             end
           rescue LoadError
             begin
               # If we couldn't load at the target/lib level check everywhere
               Cosmos.disable_warnings do
+                filename = parameters[0]
```

Comment:
This is probably a relative path.   To make @requires consistently full paths you would need to search through the require search path and try to find the file.

Reply:
Are you saying I should do a File.expand_path on filename?

Reply:
Sort of.  You will have to actually walk through the search path, and check if the File.exist? there.

Reply:
@ryanatball Can you finish this up? I'm just not tracking what you're getting at.